org.eclipse.m2e.core.ui.tests/src/org/eclipse/m2e/core/ui/tests/ConsoleTest.java

```diff
@@ -217,7 +217,8 @@ private static void assertDebugeePrintOutAndDebuggerLaunch(IDocument document, S
 		assertEquals(debugLaunchName, debugLaunch.getLaunchConfiguration()
 				.getAttribute(IJavaLaunchConfigurationConstants.ATTR_PROJECT_NAME, ""));
 		assertEquals("debug", debugLaunch.getLaunchMode());
-		assertTrue(debugLaunch.isTerminated());
+		// TODO: this fails sometimes on Mac OS, seems to be a race condition, which statement waits for its termination?
+		//assertTrue("Debug launch is not terminated yet", debugLaunch.isTerminated());
```

Comment:
That looks like something that should not be platform depended and should always work.

Reply:
yes, this is a race condition which incidentally just happened on my Mac. IMHO there is a wait missing in the test case. @HannesWell Any idea how to wait for the termination?

Reply:
Waiting for launch termination is quite ugly, you need to register a listener and then signal that it is done e.g using a countdown latch.

Reply:
Ok, can we just leave this assertion out from the test then?

Reply:
I assume there is a reason to check it is terminated, but I see some other problems here, for example that we assume there are always two launches and the first is always maven.

So I think the test might be improved in the following way:

1. Do not assert the number of launches, just assert that there are the two ones we expect (e.g. by searching for name or property)
2. If we assume they should terminate, check in a loop if they are terminated, if not do a Thread.spinWait() and check again, if not terminated after 10 seconds (or whatever seems suitable) give a meaningful exception.

Reply:
@HannesWell You wrote the code initially, so I leave that up to you to decide.

Reply:
Fixed as suggested by @laeubi in https://github.com/eclipse-m2e/m2e-core/pull/1154/commits/562182c8eb2bcdcdcf719012dd79c3aced3a69fd.

Reply:
In general I'm surprised that this is even an issue that the debugger-launch is not marked as terminated yet. At that point the maven build has completely terminated so the debugger attached to the forked surefire VM should have done the same for 'some' time too. But maybe you have a super fast computer or are just in bad luck to get hit by that race-condition.

> I assume there is a reason to check it is terminated, but I see some other problems here, for example that we assume there are always two launches and the first is always maven.
> 

It is indeed not very nice, but since the launches in the launch-manager are cleared before the build is started I think that is OK for a test-case and until now I never observed that as a problem. But of course it can be enhanced.

> Fixed as suggested by @laeubi in [562182c](https://github.com/eclipse-m2e/m2e-core/commit/562182c8eb2bcdcdcf719012dd79c3aced3a69fd).

Good. 👍🏽